### PR TITLE
Add support to list indexes

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem

The sdk lacks the ability to list indexes

## Solution

Added support to list indexes

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Added unit and integrations tests.

### CODE REVIEW

File: src/test/java/io/pinecone/PineconeIndexOperationClientTest.java [hunk 1]
```diff
@@ -221,4 +221,48 @@ public void testDescribeIndex() throws IOException {
 
         assertEquals(expectedIndexMeta.toString(), actualIndexMeta.toString());
     }
+
+    @Test
+    public void testListIndexes() throws IOException {
+        String responseBody = "[\"index1\",\"index2\"]";
```
Comment (rohanshah18) [inline src/test/java/io/pinecone/PineconeIndexOperationClientTest.java:227]: Added unit tests for indexList.size() non-zero and 0.

File: src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java [hunk 2]
```diff
@@ -41,6 +42,10 @@ public void createAndDelete() throws IOException {
         assertEquals(indexName, indexMeta.getDatabase().getName());
         assertEquals("euclidean", indexMeta.getDatabase().getMetric());
 
+        // List the index
```
Comment (rohanshah18) [inline src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java:45]: Added integration tests here.

**Review by austin-denoble — APPROVED** [review]
Looks good to me, nice work! Just one question about the loss of some imports in `PineconeIndexOperationClientTest.java`, but I'm probably just missing something. 🚢

File: src/test/java/io/pinecone/PineconeIndexOperationClientTest.java [hunk 3]
```diff
@@ -1,7 +1,5 @@
 package io.pinecone;
 
-import io.pinecone.PineconeClientConfig;
-import io.pinecone.PineconeIndexOperationClient;
```
[the remark this answers is not in the document]
Comment (rohanshah18) [reply src/test/java/io/pinecone/PineconeIndexOperationClientTest.java:4]: Yes, they weren't needed in the first place

**Pull request merged**